dev/com.ibm.ws.ejbcontainer.timer.np_fat/test-applications/NpTimersEJB.jar/src/com/ibm/ws/ejbcontainer/timer/np/ejb/XMLTxBean.java

```diff
@@ -125,6 +124,7 @@ public Date createTimer(String info) {
 
         Timer t = TimerHelper.createTimer(ivTS, DURATION, null, info, false, null);
         svTimer = t;
+        svNextTimeoutLatch = new CountDownLatch(1);
```

Comment:
I think this should go before creating the timer, otherwise it is possible the timer runs before the new latch is created, and thus it would never be timed out.

Reply:
Should be corrected in update.